File: frontend/src/microservices/advertising/objects/AdvertisementTextCandidateResponse.ts
// 本文件定义 advertising 模块的 `AdvertisementTextCandidateResponse`，用于文案候选项响应并提供 JSON 编解码。

export type AdvertisementTextCandidateResponse = {
  text: string
  emphasis: string
  seed: number
}

export const advertisementTextCandidateResponseFromJson = (json: string): AdvertisementTextCandidateResponse =>
  JSON.parse(json) as AdvertisementTextCandidateResponse

export const advertisementTextCandidateResponseToJson = (value: AdvertisementTextCandidateResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/advertising/objects/AdvertisementTextCandidateResponse.scala
// 本文件定义 advertising 模块的 AdvertisementTextCandidateResponse，作为后端与前端同名的数据模型，并提供 JSON codec。
package com.typesafe.travel.advertising.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class AdvertisementTextCandidateResponse(
    text: String,
    emphasis: String,
    seed: Int
)

object AdvertisementTextCandidateResponse:
  given sourceEncoder: Encoder[AdvertisementTextCandidateResponse] = deriveEncoder
  given sourceDecoder: Decoder[AdvertisementTextCandidateResponse] = deriveDecoder

